create documentation for operations using the new Beagle dsl (#767)
ZupIT/beagle-docs: content/pt/resources/dsl-operations/overview.md

---
title: Visão Geral
weight: 1
type: overview

> description: 'Nesta seção, você encontra a descrição completa das Operações com a utilização da nova DSL do Beagle.'

Reviewer: ```suggestion
description: 'Nesta seção, você encontra a descrição completa das Operações que utilizam a DSL do Beagle.'
```

---

---

## O que é? 

> As operações permitem que você crie aplicações com telas mais complexas por meio do seu backend com operações básicas como condicionais, soma, igualdade, etc. Dessa forma, os  componentes não necessitam de ter esse tipo de lógica dentro deles. 

Reviewer: ```suggestion
As operações permitem você criar aplicações com telas mais complexas por meio do seu backend com operações básicas como:
- Condicionais
- Soma
- Igualdade
- E outras.
A vantagem é que os componentes não precisam ter esse tipo de lógica dentro deles. 
```

## Como usar?

> De modo geral, as operações podem ser usadas quando você quer modificar os valores de um componente por meio de lógicas e operações em um momento diferente daquele em que recebeu os componentes com o JSON.
> 
> No Beagle é disponibilizado alguns operadores padrões e você pode criar suas próprias operações. 

Reviewer: ```suggestion
As operações são usadas quando você quiser modificar os valores de um componente por meio de lógicas e operações e em um momento diferente daquele que você recebeu os componentes com o JSON.

O Beagle disponibiliza alguns operadores padrões, mas você também pode criar suas próprias operações. 
```

> ##### **OBS:** 

Reviewer: ```suggestion
##### *Uso dos métodos** 
```

> - A nova DSL não implica na mudança do JSON ao renderizar os componentes;

Reviewer: ```suggestion
{{% alert color="warning" %}}
  A nova DSL não implica na mudança do JSON ao renderizar os componentes.
{{% /alert %}}

```

> - Ao utilizar os operadores com a nova DSL é necessário compreender o uso dos métodos a seguir:

Reviewer: ```suggestion
Quando você utilizar os operadores com a nova DSL, é necessário entender o uso dos métodos a seguir:
```

> **constant -** Este método é utilizado quando as operações são do tipo [Bind.Value](https://docs.usebeagle.io/v1.10/pt/api/context/overview/).
> 
> 
> **expressionOf -** Este método é utilizado quando as operações são do tipo [Bind.Expression](https://docs.usebeagle.io/v1.10/pt/api/context/overview/).
> 
> **toBindString -** Em um componente Beagle, é necessário adicionar este método após o componente para fazer a conversão da operação para o tipo Bind de String.
> 
> *Exemplo:* ```Text(sum(constant(1), constant(2)).toBindString())```
> 
> **plus -** Este método é utilizado para escrever um texto seguido de uma operação. 
> 
> **createOperation -** Por meio deste método é possível criar [operações customizadas](https://docs.usebeagle.io/v1.10/pt/api/plugins/dsl-operations/how-to-register-a-new-operation/).
> 
> *Exemplo:* ```Text(constant("Sum of 2 + 1 = ").plus(sum(constant(2), constant(1)).toBindString()))``` *ou* ```Text(constant("Sum of 2 + 1 = ") + (sum(constant(2), constant(1)).toBindString()))```

Reviewer: ```suggestion
**1.** **constant -** Este método é utilizado quando as operações são do tipo [Bind.Value](https://docs.usebeagle.io/v1.10/pt/api/context/overview/).

**2.** **expressionOf -** Este método é utilizado quando as operações são do tipo [Bind.Expression](https://docs.usebeagle.io/v1.10/pt/api/context/overview/).

**3.** **toBindString -** Em um componente Beagle, é necessário adicionar este método após o componente para fazer a conversão da operação para o tipo Bind de String.

*Exemplo:* ```Text(sum(constant(1), constant(2)).toBindString())```

**4.** **plus -** Este método é utilizado para escrever um texto seguido de uma operação. 

**5.** **createOperation -** Por meio deste método é possível criar [operações customizadas](https://docs.usebeagle.io/v1.10/pt/api/plugins/dsl-operations/how-to-register-a-new-operation/).

*Exemplo:* ```Text(constant("Sum of 2 + 1 = ").plus(sum(constant(2), constant(1)).toBindString()))``` *ou* ```Text(constant("Sum of 2 + 1 = ") + (sum(constant(2), constant(1)).toBindString()))```
```
